iothub/service/src/IotHubServiceClient.cs

```diff
@@ -203,6 +209,42 @@ public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHu
         /// </summary>
         /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d"/>.
         public MessagesClient Messages { get; protected set; }
+        
+        /// <summary>
+        /// Requests connection string for the built-in Event Hubs messaging endpoint of the associated IoT hub.
+        /// </summary>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>A connection string which can be used to connect to the Event Hubs service and interact with the IoT hub messaging endpoint.</returns>
+        /// <exception cref="InvalidOperationException">The Event Hubs host information was not returned by the IoT hub service.</exception>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-endpoints"/>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-read-builtin"/>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-amqp-support#receive-telemetry-messages-service-client"/>
+        public Task<string> GetEventHubCompatibleConnectionStringAsync(CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(_connectionString))
+            {
+                throw new InvalidOperationException("This overload must only be called if the client was initialized with a connection string.");
+            }
+
+            return EventHubConnectionStringBuilder.GetEventHubCompatibleConnectionStringAsync(_connectionString, cancellationToken);
+        }
+
+        /// <summary>
+        /// Requests connection string for the built-in Event Hubs messaging endpoint of the associated IoT hub.
+        /// </summary>
```

Comment:
Remarks: Use this overload when retrieving the Event Hubs compatible connection string for an IotHubServiceClient that was initialized using a method other than IoT hub connection string.